src/core_plugins/vega/public/vega_request_handler.js

```diff
@@ -0,0 +1,18 @@
+import { VegaParser } from './vega_view/vega_parser';
+import { dashboardContextProvider } from 'plugins/kibana/dashboard/dashboard_context';
+
+export function VegaRequestHandlerProvider(Private, es, timefilter, serviceSettings) {
+
+  const dashboardContext = Private(dashboardContextProvider);
+
+  return {
+
+    name: 'vega',
+
+    handler(vis) {
```

Comment:
Is it possible to memoize the backend call based on the `vis.params.spec`-configuration? ie.  If there isn't a change in the body.data, we can return an earlier response.

Right now, we do a backend call to the server, even if we are just updating styling parameters.

Reply:
+1, this is a must have, calls to backend should only happen:
- if we have a valid data request (query validation)
- if our request is different than the previous one or some time has elapsed

Reply:
Yep, the caching is already done - https://github.com/elastic/kibana/pull/15014/files#diff-140acbabe1232e5ba0ae5963de4b9865

The problem is that if the query contains relative time filter (and most of them probably would), each request is slightly different.  We could force timestamp rounding, e.g. to the nearest 5 seconds (that's the smallest value of the autoupdate). Not very valuable in terms of savings - debouncing graph refreshes on keystrokes would provide similar levels of optimization. Also, user has the option of not using auto-update. So really this is an optimization just for when a user wants to design a new graph, and wants to enable auto-update, and for graphs that are not time-based.

Reply:
Update: Now the query is generated using timeCache object, which will return unchanged time for up to N seconds (hardcoded at 3s).  The time cache is ignored if timefilter min or max change more than 3s.  This way all queries will be generated with identical timestamps. Also, the timecache is global, ensuring that multiple visualization instances with the same query will also generate the same request, and will reuse the cache (even if the query hasn't returned yet)